### ISSUE TITLE
Reset on prefilledProps change?

### ISSUE THREAD

**Issue by eric-burel** [VulcanJS/Vulcan#2283]
Right now the only way to pass down fields to the form from a parent component is using `prefilledProps`. However a change in prefilledProps will trigger a form reset, so we can't "hot" update the form from the parent without losing the data entered by the user.

We had a discussion about this: https://github.com/VulcanJS/Vulcan/pull/2127, I had this question:

> Props that should trigger a state reset when modified: 'collection', 'collectionName', 'typeName', 'document', 'schema', 'currentUser'.
> Props I'd specifically like to have your opinion about: 'fields', 'removeFields', 'prefilledProps', 'client'. I think fields should reset, prefilledProps should not, and I don't know about the others.

The question still holds, especially for prefilledProps. Maybe we could also add different props depending on the use case.
Tagging @jimrandomh @justinr1234 because you were involved in the previous PR about state reset

**Comment by jimrandomh** [comment]
If `prefilledProps` changes, the possibilities are:

 * The change is spurious (eg deep-equal objects comparing shallow-unequal). This is the common case, and in this case, resetting is bad.
 * The change is real, and the user has already edited the form. This is a weird rare case. In this case, resetting the form would lose their data, so resetting is bad.
 * The change is real, and the user hasn't touched the form yet. This is a weird rare case. In this case, resetting is good.

Overall, I'd say resetting the form on a `prefilledProps` change is a bad idea.

There are cases where you want want to respond to something happening outside a form, by changing the value of a specific form field. This requires something more granular than resetting the form as a whole; you would need to break option the form abstraction and deal with individual fields, probably by implementing the form the old fashioned way, rather than using `vulcan-forms`.

**Comment by SachaG** [comment]
Yeah it does sound like changing prefilledProps *shouldn't* reset the form. I think the use case for resetting is for example a list of items with an edit form on the same screen. When selecting a different item to edit you do want the form next to the list to be reset, as it almost becomes a different form altogether in the user's mental model. 

Maybe we could only reset the form when the document's `_id` changes? That might be a bit arbitrary but it does work for that use case.
